source/modulo_components/include/modulo_components/ComponentInterface.hpp

```diff
@@ -195,6 +194,20 @@ ComponentInterface<NodeT>::ComponentInterface(
         return this->on_set_parameters_callback(parameters);
       });
   this->add_parameter("period", 1.0, "The time interval in seconds for all periodic callbacks", true);
+  this->add_parameter("has_tf_listener", false, "Indicate if the component needs to instantiate a tf listener", true);
+  this->add_parameter("has_tf_broadcaster", false, "Indicate if the component needs to instantiate a tf braodcaster", true);
+
+  this->period_ = NodeT::get_parameter("period");
+  this->has_tf_listener_ = NodeT::get_parameter("has_tf_listener");
+  this->has_tf_broadcaster_ = NodeT::get_parameter("has_tf_broadcaster");
+
+  if (has_tf_listener_.as_bool()) {
+    this->tf_buffer_ = std::make_shared<tf2_ros::Buffer>(this->get_clock());
+    this->tf_listener_ = std::make_shared<tf2_ros::TransformListener>(*this->tf_buffer_);
+  }
+  if (has_tf_broadcaster_.as_bool()) {
+    this->tf_broadcaster_ = std::make_shared<tf2_ros::TransformBroadcaster>(this->shared_from_this());
+  }
```

Comment:
All this should be removed, this was not part of your original PR

Reply:
It might have been actually but should not be here anyway. This should be added later in another PR and updated with the new parameters accessors.

Reply:
No this was removed on purpose

Reply:
ok

Reply:
alright yes that makes a lot of sense